src/core/layout/qgslayoutitemmap.cpp

```diff
@@ -447,7 +447,7 @@ void QgsLayoutItemMap::setMapRotation( double rotation )
   mMapRotation = rotation;
   mEvaluatedMapRotation = mMapRotation;
   invalidateCache();
-  emit mapRotationChanged( rotation );
+  emit mapRotationChanged( rotation + itemRotation() );
```

Comment:
The mapRotationChanged signal is used quite heavily in other areas of code outside of the north arrow, so I'm reluctant to change it's current behavior here. I'd suggest instead hooking into the existing QgsLayoutItem::rotationChanged signal, i.e. by making the picture item listen out for both the mapRotationChanged AND rotationChanged signals (i.e. adding new connections to rotationChanged wherever QgsLayoutItemPicture currently uses mapRotationChanged )

Also note that you'll need to update QgsLayoutItemPicture::updateMapRotation() -- it doesn't use the rotation value from the mapRotationChanged signal in any case, so it needs to be updated to base the picture orientation on mRotationMap->mapRotation() + mRotationMap->rotation().

Reply:
I'm confused as to why the value provided was useless, I saw the function and thought it was overloaded somewhere else to accept the given value and couldn't find that overloaded method. + the holiday lack of sleep doesn't help. I was debating simply changing the function at first but didn't want to alter some core component if it could be prevented.

Reply:
It's not useless, it's just not used by the north arrow code. Check the QgsLayoutItemPicture::updateMapRotation() slot, it doesn't have an argument accepting the rotation value used in the signal.